### PR TITLE
initialize default service & account/chain types with templates

### LINKED ISSUE #1003: [Services/maker] init should generate the necessary default definition files from go templates
@zramsay commented on [Tue Oct 04 2016](https://github.com/eris-ltd/eris-services/issues/55)
- [x] remove third party service definition files & Dockerfiles
- [x] keep anything ops related for cli but move to `/hello-eris/dockerfiles`
- [x] have cli generate service definition files & chain/account-type (from cm) using go templates

### PR DESCRIPTION
**Pull request by zramsay**
- closes #1003 and #917
- includes plenty of additional small fixes
- removes:
```
[location]
dockerfile = ""
repo = ""
website = ""
```
from service definition files.

### CODE REVIEW

File: loaders/services.go [hunk 1]
```diff
@@ -61,10 +61,9 @@ func MockServiceDefinition(servName string) *definitions.ServiceDefinition {
 // MarshalServiceDefinition converts a Viper configuration structure to a
 // service definition one; it can return marshalling errors.
 func MarshalServiceDefinition(serviceConf *viper.Viper, srv *definitions.ServiceDefinition) error {
-	err := serviceConf.Unmarshal(srv)
-	if err != nil {
-		// Vipers error messages are atrocious.
```
Comment (zramsay) [inline loaders/services.go]: they've become  quite useful now

File: chains/chains_test.go [hunk 2]
```diff
@@ -17,10 +17,7 @@ import (
 	"github.com/eris-ltd/eris/util"
 )
 
-var (
-	erisDir   = filepath.Join(os.TempDir(), "eris")
-	chainName = "test-chain"
-)
+var chainName = "test-chain"
```
Comment (VoR0220) [inline chains/chains_test.go]: ++ much more simple

File: config/config.go [hunk 3]
```diff
@@ -78,29 +78,25 @@ func New(writer, errorWriter io.Writer) (*Config, error) {
 // LoadViper reads the definition file pointed to by
 // the definitionPath path and definitionName filename.
 func LoadViper(definitionPath, definitionName string) (*viper.Viper, error) {
-	var errKnown string
-	switch definitionPath {
-	case ServicesPath:
-		errKnown = fmt.Sprintf(`
-
-List available definitions with the [eris %s ls --known] command`, filepath.Base(definitionPath))
-	}
```
Comment (VoR0220) [inline config/config.go]: good. This was ugly.

File: definitions/account_types.go [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-	Tokens      int            `mapstructure:"default_tokens" json:"default_tokens" yaml:"default_tokens" toml:"default_tokens"`
-	ToBond      int            `mapstructure:"default_bond" json:"default_bond" yaml:"default_bond" toml:"default_bond"`
-	Perms       map[string]int `mapstructure:"perms" json:"perms" yaml:"perms" toml:"perms"`
+	Name          string         `mapstructure:"name" json:"name" yaml:"name" toml:"name"`
+	Description   string         `mapstructure:"description" json:"description" yaml:"description" toml:"description"`
+	TypicalUser   string         `mapstructure:"typical_user" json:"typical_user" yaml:"typical_user" toml:"typical_user"`
+	DefaultNumber int            `mapstructure:"default_number" json:"default_number" yaml:"default_number" toml:"default_number"`
+	DefaultTokens int            `mapstructure:"default_tokens" json:"default_tokens" yaml:"default_tokens" toml:"default_tokens"`
+	DefaultBond   int            `mapstructure:"default_bond" json:"default_bond" yaml:"default_bond" toml:"default_bond"`
+	Perms         map[string]int `mapstructure:"perms" json:"perms" yaml:"perms" toml:"perms"`
```
Comment (VoR0220) [inline definitions/account_types.go]: better names ++

File: initialize/default_compilers.go [hunk 5]
```diff
@@ -1,44 +0,0 @@
...
-environment    = [  ]
-
-[maintainer]
-name = "Monax Industries"
-email = "support@monax.io"
-
-[location]
-repository = "https://github.com/eris-ltd/eris-compilers"
-website = "https://monax.io/docs/documentation/compilers"
-`
```
Comment (VoR0220) [inline initialize/default_compilers.go]: going to have to keep this in mind for my compilers redo.

**Comment by VoR0220** [comment]
@zramsay needs a rebase

File: initialize/default_files.go [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+	case "compilers":
+
+		serviceDefinition.Name = "compilers"
+		serviceDefinition.Description = `Monax's Solidity Compiler Server.
+
+This eris service compiles smart contract languages.`
+		serviceDefinition.Status = "beta"
+		serviceDefinition.Service.Image = path.Join(version.DefaultRegistry, version.ImageCompilers)
+		serviceDefinition.Service.AutoData = true
+		//serviceDefinition.Service.Ports = []string{`"9090:9090"`}
```
Comment (zramsay) [inline initialize/default_files.go]: ping @VoR0220 this will need to be uncommented when sync'ing up with a recent compilers PR of yours

Comment (VoR0220) [inline initialize/default_files.go]: Si. Thanks for pointing that out.

File: initialize/default_files.go [hunk 7]
```diff
@@ -0,0 +1,431 @@
...
+}
+
+const accountTypeDefinitionGeneral = `
+name = "{{ .Name}}"
+
+description = """
+{{ .Description}}
+"""
+
+typical_user = """
```
Comment (benjaminbollen) [inline initialize/default_files.go]: What does 'typical' mean here; would 'user' suffice?

File: loaders/services.go [hunk 8]
```diff
@@ -61,10 +61,9 @@ func MockServiceDefinition(servName string) *definitions.ServiceDefinition {
 // MarshalServiceDefinition converts a Viper configuration structure to a
 // service definition one; it can return marshalling errors.
 func MarshalServiceDefinition(serviceConf *viper.Viper, srv *definitions.ServiceDefinition) error {
-	err := serviceConf.Unmarshal(srv)
```
Comment (benjaminbollen) [inline loaders/services.go]: @silasdavis we should follow the paradigm of tooling here for e-db too

File: initialize/default_files.go (hunk 7 above)
Comment (zramsay) [inline initialize/default_files.go]: "who the usual user of this account is". I quite like it

**Pull request merged**